plugin/src/main/java/org/opensearch/ml/action/connector/GetConnectorTransportAction.java

```diff
@@ -77,6 +80,15 @@ protected void doExecute(Task task, ActionRequest request, ActionListener<MLConn
                         ensureExpectedToken(XContentParser.Token.START_OBJECT, parser.nextToken(), parser);
                         Connector mlConnector = Connector.createConnector(parser);
                         mlConnector.removeCredential();
+                        if (!Objects.equals(tenantId, mlConnector.getTenantId())) {
```

Comment:
Is the expectation that tenant ID will be null in cases of independent node settings is disabled?

Reply:
Yes